newton/_src/geometry/narrow_phase.py

```
@@ -1912,6 +2008,35 @@ def launch_custom_write(
                 writer_data,
             )
 
+        # Verify no collision pipeline buffers overflowed
+        wp.launch(
+            kernel=verify_narrow_phase_buffers,
+            dim=[1],
+            inputs=[
+                num_candidate_pair,
```

Ruff (0.15.0)

[error] Undefined name `num_candidate_pair`

(F821)